pkg/cmd/cli/cmd/wrappers.go

```diff
@@ -185,28 +185,34 @@ const (
 
   # Switch to raw terminal mode, sends stdin to 'bash' in ruby-container from pod 'mypod' and sends stdout/stderr from 'bash' back to the client
   %[1]s exec mypod -c ruby-container -i -t -- bash -il`
+)
 
-	completionLong = `This command prints shell code which must be evaluation to provide interactive
-completion of openshift cli commands.`
+func NewCmdCompletion(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	cmdHelpString := "openshift cli"
+	cmdHelpName := fullName
+
+	if strings.HasSuffix(fullName, "admin") || strings.HasSuffix(fullName, "oadm") {
+		cmdHelpString = "openshift admin"
+	} else if strings.HasSuffix(fullName, "completion") {
+		cmdHelpName = "openshift"
+		cmdHelpString = "openshift"
+	}
 
-	completionExample = `  # Generate the openshift cli completion code for bash
-  source <(oc completion bash)
+	cmd := kcmd.NewCmdCompletion(f.Factory, out)
+	cmd.Long = `This command prints shell code which must be evaluated to provide interactive
```

Comment:
Use `fmt.Printf` and keep the messages in the `completionLong` and `completionExample` const.